mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/tabs/subscribers/filter.tsx

```diff
@@ -29,19 +36,79 @@ function getStatusOptions() {
 function getStepOptions(automation: Automation) {
   const steps = Object.values(automation.steps)
     .map((step) => {
-      if (step.id === 'root') {
+      if (step.type !== 'action') {
         return null;
       }
-      const label = select(editorStoreName).getRegistryStep(step.key)?.name;
+      const name = select(editorStoreName).getRegistryStep(step.key)?.name;
       return {
-        value: step.id,
-        label,
+        id: step.id,
+        key: step.key,
+        name,
+        children: [],
       };
     })
     .filter((value) => value !== null)
-    .sort(sortByLabelCallback);
+    .sort(sortByNameCallback);
 
-  return [{ label: __('All', 'mailpoet'), value: '' }].concat(steps);
+  const stepTypes: Record<string, MultiSelectOption> = {};
+  for (let i = 0; i < steps.length; i += 1) {
+    const s = steps[i];
+    const key = s.key;
+    if (stepTypes[key] === undefined) {
+      stepTypes[key] = {
+        id: '',
+        name: s.name,
+        children: [],
+      };
+    }
+    stepTypes[key].id =
+      stepTypes[key].id.length > 0 ? `${stepTypes[key].id},${s.id}` : s.id;
```

Comment:
As this is a temporary solution, I did not put too much thought into it. I think for now it's okay, as it will be probably redone with the new woo component.